giant/plugins/client/objc_ios/objc_ios/objc_ios/templates/Utility/JSONable.m.jinja

```diff
@@ -62,6 +62,13 @@
     return [NSDate {{ class_prefix }}_dateWithISO8601String:(NSString *)json];
 }
 
++ (instancetype)fromJSON:(id<{{ class_prefix }}JSONable>)json withFormat:(NSString*)format
+{
+    NSDateFormatter* dateFormatter = [NSDateFormatter new];
```

Comment:
So this will construct the dateformatter every time we parse a JSON element. I'd prefer to have a local static dictionary of formatters to lazy load them.

Maybe something like this:

```
static NSMutableDictionary* formatters = [NSMutableDictionary new];
NSDateFormatter* formatter = formatters[format];
if(formatter == nil) {
    formatter = [NSDateFormatter new];
    formatter.dateFormat = format;
    formatters[format] = formatter
}
return [formatter dateFromString:(NSString*)json];
```